src/main.cpp

```diff
@@ -4790,6 +5054,12 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
 
         pfrom->fClient = !(pfrom->nServices & NODE_NETWORK);
 
+        if((pfrom->nServices & NODE_WITNESS))
+        {
+            LOCK(cs_main);
+            State(pfrom->GetId())->fHaveWitness = true;
```

Comment:
Not sure how useful this is. `(pfrom->nServices & NODE_WITNESS)` is 33 characters, `State(pfrom->GetId())->fHaveWitness` is 35 characters. It's easier and quicker and uses up less memory to type the thing this variable returns.